userspace/libsinsp/runc.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright (C) 2013-2019 Draios Inc dba Sysdig.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "runc.h"
+
+#include <cstring>
+
+#include "sinsp.h"
+#include "sinsp_int.h"
+
+namespace {
+
+const size_t CONTAINER_ID_LENGTH = 64;
+const size_t REPORTED_CONTAINER_ID_LENGTH = 12;
+const char* CONTAINER_ID_VALID_CHARACTERS = "0123456789abcdefABCDEF";
+
+static_assert(REPORTED_CONTAINER_ID_LENGTH <= CONTAINER_ID_LENGTH, "Reported container ID length cannot be longer than actual length");
+
+bool detect_one_container_id(const std::string& cgroup, const std::string& prefix, const std::string& suffix, std::string& container_id)
+{
+	size_t pos = cgroup.rfind(prefix);
+	if (pos == std::string::npos)
+	{
+		return false;
+	}
+	pos += prefix.size();
+
+	size_t pos2 = cgroup.rfind(suffix);
+	if (pos2 == std::string::npos)
+	{
+		return false;
+	}
+
+	if (pos2 - pos != CONTAINER_ID_LENGTH)
+	{
+		return false;
+	}
+
+	static_assert(std::string::npos == (size_t)(-1), "std::string::npos must be largest valid size_t");
+	size_t invalid_ch_pos = cgroup.find_first_not_of(CONTAINER_ID_VALID_CHARACTERS, pos);
+	if (invalid_ch_pos < CONTAINER_ID_LENGTH)
```

Comment:
just match the ID against [0-9a-fA-F]{64} ?

Reply:
This is literally what this code does. Just not using regexes.